### PR TITLE
Refactor file structure, enhance backup manager functionality and fix permissions issues

### LINKED ISSUE #117: [Feature Request] Allow to run with userID other than 1000:1000
### Have you read the Important information text above

- [X] Yes i did

### Describe the feature

Currently we are forced to run the survice with USER and GROUP ID 1000 which is ok if i run this on an environment where this user is availiable to me. 

However if not (or if i want to run it with another userid) there is currently no way to do this easily (this is kind of related to the used baseimage which is build with default PUID=1000

Also it is not a very good approach to require the user to run chmod 777 on the data directory (which will not work if one is using an NFS share for game files mounted as docker volume directly)

Allowing correct user IDs and using the non root base container is a workaround for this though. 
Thats what i did on my infra in a sloppy way. 

If you find this valuable im happy to submit a PR that adds this functionality to your image/stack

### Additional information

- [X] Would you be willing to help implement this feature?

### Final checks

- [X] I understand that this is a feature request function. I also unterstand that if i use this to report a bug report, that my issue might be un-answered closed.

### LINKED ISSUE #156: [Bug Report] RCON config file doesn't update from docker env variables
### Have you read the Important information text above

- [X] Yes i did

### Current behavior

Using the latest few images, I am seeing this error in the logs

```
time="2024-01-31T09:00:05-08:00" level=info msg="cli: execute: auth: rcon: dial tcp: address tcp/###RCON_PORT###: unknown port" channel=stderr iteration=0 job.command=/backupmanager.sh job.position=0 job.schedule="0 * * *
```

I'm also getting the same errors when I run it via command line within the container

### Desired behavior

The startup scripts replace the placeholder values with the environment variables if the RCON functionality is enabled

### Links to screenshots

_No response_

### To Reproduce

1. Run latest image in docker with `RCON_ENABLED` set to `true`
2. Wait for the container to startup
3. Look at logs and see errors

### Software setup

- OS: Ubuntu
- Docker: Docker version 25.0.1, build 29cf629

### Hardware setup

- vCPU: unrestricted
- RAM: unrestricted
- Disk: unrestricted

### Additional context

I replaced the values manually to see if it works, but now I am getting a connection refused error, so I'm not sure if there are other errors?

### PR DESCRIPTION
**Pull request by thejcpalma**
# Enhancements and Refactoring

## Summary

This PR introduces several enhancements and changes aimed at improving the organization, readability, and functionality of the project. It includes restructuring the file structure, refactoring the code, improving log output, enhancing the backup manager, and updating the Dockerfile and README.md.

## Motivation

The motivation behind these changes were to enhance the project's already great structure and functionality. I wanted to do linting on the code to ensure it adheres to best practices and standardization, allowing for the addition of the flow to check shell linting.
I also wanted to improve the log output for better readability, hence the new log messages (still to improve). 
Also, to enhance the backup manager to accept arguments for multiple backup operations. 
And to fix the permissions issues with the Dockerfile and the volumes, leading to users having to run `chmod` or `chown` on the host. (Effectively achieving what I wanted in #147)

## Changes

- Restructured the file structure for better organization.
- Refactored code to best practices and to standardize it.
- Improved output of logs for readability (with colors too!).
- Enhanced backup manager to accept arguments for multiple backup operations.
- Added ability to run backup manager as a pseudo binary (symlink to /usr/local/bin/).
- Removed `VOLUME` and `USER` directives from Dockerfile to fix permissions issues.
- Added a new entrypoint script to handle permissions and launch servermanager.sh as steam user.
- Updated README.md to reflect changes and improve usability.
- Added the setup of `rcon.yaml` to be enforced if `RCON_EBABLED=true` and `RCON_PORT=<rcon_port_number>` and `ADMIN_PASSWORD=<your_admin_pw>` all exist and are set, independently of the value in `SERVER_SETTING_MODE`.

## Testing

The changes were validated by running the project locally multiple times with different setups of docker compose, volumes and named volumes and verifying permissions and correct ownership of the main script running on the image.
The refactored code was tested with [ShellCheck](https://marketplace.visualstudio.com/items?itemName=timonwong.shellcheck).
The improved log output was checked for better readability on both live logging and inspection.
The enhanced backup manager was tested with multiple backup operations, during multiple stages of the container, to ensure the best usage and minimize risk scenarios.
The updated Dockerfile was used to build a Docker image and run a container to confirm the fixed permissions issues.
The new entrypoint script was executed to verify it handles permissions and launches servermanager.sh as the steam user.
The updated README.md was reviewed to ensure it accurately reflects the changes and improves usability.

## Additional Information

These changes collectively aim to improve the project's overall quality and usability.
Also complete/fix #147.
Fixes #156.
Fixes #117.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+
+if [[ "${PUID}" -eq 0 ]] || [[ "${PGID}" -eq 0 ]]; then
+    pp --warning "[WARNING] Running as root is not supported, please fix your PUID and PGID!\n"
+    exit 1
+elif [[ "$(id -u steam)" -ne "${PUID}" ]] || [[ "$(id -g steam)" -ne "${PGID}" ]]; then
+    pp --info "Old steam user UID was '$(id -u steam)' and GID was '$(id -g steam)'\n"
+    pp --info "Setting steam user UID to '${PUID}' and GID to '${PGID}'\n"
+    usermod -o -u "${PUID}" steam
+    groupmod -o -g "${PGID}" steam
```
Comment (masonen) [inline entrypoint.sh:14]: i like that approach. my workaround was similar but thats even better.

Comment (thejcpalma) [reply entrypoint.sh:14]: Using another popular repos to see how people are handling their problems and seeing how they approach it, but always with a foot back. I tested it, checked for security issues it might create, and only then I used this approach.
Wouldn't want to introduce a vulnerability in this beautiful project

File: scripts/backupmanager.sh [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+    echo "  num_to_keep         (optional) The number of the most recent backup files to keep."
+    echo "                                 If not provided, the value of the BACKUP_RETENTION_AMOUNT"
+    echo "                                 environment variable will be used if it exists and is not empty."
+    echo "                                 Defaults to the 5 most recent backups otherwise."
+}
+
+
+# Function to parse input arguments
+function parse_arguments() {
+    if [ ${#} -lt 1 ]; then
```
Comment (jammsen) [inline scripts/backupmanager.sh:55]: This check should have been done before entering the function.
This should be in initializeBackupManager and if nothing is valid show usage.
This function jumping is un-neccesary and disturbs the flow of reading code, because you have to jump every 2 seconds to antoher function, you miss overview and context that way. KISS principle please (Keep it simple and stupid).

File: scripts/backupmanager.sh [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+SAVE_DIR=${GAME_SAVE_PATH} # Dir where the game save files are stored
+BACKUPS_TO_KEEP=${BACKUP_RETENTION_AMOUNT} # Number of backup files to keep
+
+
+# Function to print usage information
+function print_usage() {
+    script_name=$(basename "$0")
+    echo "Usage:"
+    echo "  ${script_name} [backup_directory] -c|--create"
+    # echo "  ${script_name} [backup_directory] -r|--restore [backup_name]"
```
Comment (jammsen) [inline scripts/backupmanager.sh:20]: No commented update code in production, if its done, put it in, if not, then remove it.

Only # comments for context and links and sources.

File: scripts/backupmanager.sh [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+    echo "Arguments:"
+    echo "  backup_directory    (optional) The directory where the backup files are stored."
+    echo "                                 If not provided, the value of the BACKUP_PATH"
+    echo "                                 environment variable will be used if it exists"
+    echo "                                 and is not empty."
+    # echo "  backup_name         (optional) The name of the backup file to restore from."
+    # echo "                                 If not provided, the value of the BACKUP_FILE"
+    # echo "                                 environment variable will be used if it exists"
+    # echo "                                 and is not empty."
+    echo "  num_entries         (optional) The number of backup files to list."
```
Comment (jammsen) [inline scripts/backupmanager.sh:44]: See other mentions.

File: scripts/backupmanager.sh [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+function parse_arguments() {
+    if [ ${#} -lt 1 ]; then
+        pp --warning "> Not enough arguments.\n"
+        print_usage
+        exit 1
+    fi
+
+    # Check if the backup directory was given as an argument
+    case "$2" in 
+        -c|--create|-r|--restore|-l|--list)
```
Comment (jammsen) [inline scripts/backupmanager.sh:63]: Why have 6 alias to check?

File: scripts/backupmanager.sh [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+    script_name=$(basename "$0")
+    echo "Usage:"
+    echo "  ${script_name} [backup_directory] -c|--create"
+    # echo "  ${script_name} [backup_directory] -r|--restore [backup_name]"
+    echo "  ${script_name} [backup_directory] -l|--list [num_entries]"
+    echo "  ${script_name} [backup_directory] -cl|--clean [num_to_keep]"
+    echo "  ${script_name} -h|--help"
+    echo ""
+    echo "Options:"
+    echo "  -c, --create                    Create a backup"
```
Comment (jammsen) [inline scripts/backupmanager.sh:26]: I favor --readable-and-understandable-options not -c. The heck is -c? What manual do i need to read now? 😎 
Just get rid of the "-y" options, doesnt improove readability. Thats very old style of coding.

File: scripts/backupmanager.sh (hunk 5 above)
Comment (jammsen) [reply scripts/backupmanager.sh:63]: See above

File: scripts/backupmanager.sh [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+        *)
+            pp --error "Illegal option '${1}'\n"
+            print_usage
+            exit 1
+            ;;
+    esac
+}
+
+function check_required_directories() {
+    # Exit if backup dir or game save dir don't exist
```
Comment (jammsen) [inline scripts/backupmanager.sh:130]: The code already describes that, why is that here?

File: scripts/backupmanager.sh [hunk 8]
```diff
@@ -0,0 +1,336 @@
...
+        pp --error "> Please provide the backup directory as an argument or set the BACKUP_PATH environment variable.\n"
+        pp --error "> Exiting...\n"
+        exit 1
+    elif [ -z "${SAVE_DIR}" ]; then
+        pp --error "> GAME_SAVE_PATH environment variable not set.\n"
+        pp --error "> Exiting...\n"
+        exit 1
+    fi
+
+    # If vars are set, check if the directories exist
```
Comment (jammsen) [inline scripts/backupmanager.sh:143]: Same here

File: scripts/backupmanager.sh [hunk 9]
```diff
@@ -0,0 +1,336 @@
...
+        pp --error "[ERROR] Backup directory '${BACKUP_DIR}' does not exist.\n"
+        exit 1
+    elif [ ! -d "${SAVE_DIR}" ]; then
+        pp --error "[ERROR] Game save directory '${SAVE_DIR}' does not exist.\n"
+        exit 1
+    fi
+}
+
+
+### Backup Functions
```
Comment (jammsen) [inline scripts/backupmanager.sh:154]: Random text here, not needed 🤣

File: scripts/backupmanager.sh [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+    if [ -z "${GAME_PATH}" ]; then
+        pp --error "> RESTORE_FILE_PATH environment variable not set. Exiting...\n"
+        exit 1
+    fi
+    
+    DATE=$(date +%Y%m%d_%H%M%S)
+    TIME=$(date +%H-%M-%S)
+
+    #Send message to gameserver via RCON if enabled
+    rc "broadcast ${TIME}-Backup_in_progress" "> Broadcasting server shutdown via RCON...\n"
```
Comment (jammsen) [inline scripts/backupmanager.sh:170]: rc? What? Another alias? Come on .... the command is either rconcli or rcon, do you really need to short that?  🤣 Dont be lazy, you can copy paste, earn your medal 🥇

File: scripts/backupmanager.sh [hunk 11]
```diff
@@ -0,0 +1,336 @@
...
+
+    # Create backup dir if it doesn't exist
```
Comment (jammsen) [inline scripts/backupmanager.sh:172]: remove that

File: scripts/backupmanager.sh [hunk 12]
```diff
@@ -0,0 +1,336 @@
...
+    mkdir -p "${BACKUP_DIR}"
+
+    #Send message to gameserver via RCON if enabled
+    rc 'broadcast Saving world...' "> Sending message to gameserver via RCON..."
```
Comment (jammsen) [inline scripts/backupmanager.sh:176]: This wont work
broadcast cant have " ", meaning spaces, there is a reason i wrote "stuff-likes-this-to-the-gameserver".

File: scripts/backupmanager.sh [hunk 13]
```diff
@@ -0,0 +1,336 @@
...
+    rc 'save'
+    rc 'broadcast Done!' "> Broadcast via RCON complete!"
+
+
+    # Create backup
+    tar cfz "${BACKUP_DIR}/saved-${DATE}.tar.gz" -C "${GAME_PATH}" Saved/
```
Comment (jammsen) [inline scripts/backupmanager.sh:182]: This will create bad tgz files, be in the directory you want to work in to not create shadowy structures that are not needed inside the archive 👎

File: scripts/backupmanager.sh [hunk 14]
```diff
@@ -0,0 +1,336 @@
...
+}
+
+function list_backups() {
+    local num_backup_entries=${1:-""}
+
+    pp --success ">>> Listing backups:\n\n"
+
+    # Check num_backup_entries argument
+    # if argument exists and is not a positive integer, print usage message and exit
+    if [[ -n "$num_backup_entries" && ! "$num_backup_entries" =~ ^[0-9]+$ ]]; then
```
Comment (jammsen) [inline scripts/backupmanager.sh:195]: You defined a function called parse_arguments, why do this here an not there?!

File: scripts/backupmanager.sh [hunk 15]
```diff
@@ -0,0 +1,336 @@
...
+    fi
+
+    # Check if there are any files in the backup directory
+    if [ -z "$(ls -A "$BACKUP_DIR")" ]; then
+        pp --info "No backups in the backup directory $BACKUP_DIR.\n"
+        exit 0
+    fi
+
+    # get file list
+    files=$(find "$BACKUP_DIR" -maxdepth 1 -type f -name "*.tar.gz" | sort -r)
```
Comment (jammsen) [inline scripts/backupmanager.sh:213]: ls -1t saved-*.tar.gz does the same but faster and in 1 command

File: scripts/backupmanager.sh [hunk 16]
```diff
@@ -0,0 +1,336 @@
...
+    if [ -z "$num_backup_entries" ]; then
+        # if num_backup_entries is not set, use all files
+        file_list=$files
+    else
+        # if num_backup_entries is set, use the first num_backup_entries files
+        file_list=$(echo "$files" | head -n "$num_backup_entries")
+    fi
+
+    # print file list (currently using date from file name but can use creation date from 'stat')
+    for file in $file_list; do
```
Comment (jammsen) [inline scripts/backupmanager.sh:225]: Why do this so complicated?
Why parse something?
ls and the filename is all you need, aint it?

File: scripts/backupmanager.sh [hunk 17]
```diff
@@ -0,0 +1,336 @@
...
+}
+
+function clean_backups() {
+    local num_files_to_keep=${1:-${BACKUPS_TO_KEEP}}
+
+    pp --success ">>> Backup cleaning started..\n\n"
+
+    if [[ -z "$num_files_to_keep" ]]; then
+        pp --warning "> Number of backups to keep is empty. Using default value of 30.\n"
+        num_files_to_keep=30
```
Comment (jammsen) [inline scripts/backupmanager.sh:260]: Why hardcode 30?
Use default-set of ENV in Dockerfile

File: scripts/backupmanager.sh [hunk 18]
```diff
@@ -0,0 +1,336 @@
...
+
+    # Check if there are any files in the backup directory
+    if [ -z "$(ls -A "${BACKUP_DIR}")" ]; then
+        pp --info "> No files in the backup directory ${BACKUP_DIR}. Exiting...\n\n"
+        exit 0
+    fi
+
+    pp --info "> Keeping latest ${num_files_to_keep} backups.\n"
+    
+    files_to_delete=$(find "${BACKUP_DIR}" -maxdepth 1 -type f -name "saved-*.tar.gz" \
```
Comment (jammsen) [inline scripts/backupmanager.sh:277]:
```
ls -1t saved-*.tar.gz | tail -n +$(($BACKUP_RETENTION_AMOUNT_TO_KEEP + 1)) | xargs -d '\n' rm -f --
```
Does that same but faster and in 1 line instead of 10-15

Not every old code i had is bad you know?

File: scripts/backupmanager.sh [hunk 19]
```diff
@@ -0,0 +1,336 @@
...
+        echo "$files_to_delete" | xargs -d '\n' rm -f --
+        pp --warning "> Deleted ${num_files_to_delete} file(s).\n"
+    else
+        pp --info "> No files to delete.\n"
+    fi
+
+    pp --success "\n>>> Cleaning finished!\n"
+}
+
+# function restore_backup() {
```
Comment (jammsen) [inline scripts/backupmanager.sh:296]: Why is this commented out? Wasnt this the reason for all this work in the first place? 😕

File: scripts/clean_backups.sh [hunk 20]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+if [ $# -gt 1 ]; then
```
Comment (jammsen) [inline scripts/clean_backups.sh:3]: Thats dublicate error-handling, why?

File: scripts/create_backup.sh [hunk 21]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+if [ $# -gt 0 ]; then
```
Comment (jammsen) [inline scripts/create_backup.sh:3]: Same here

File: scripts/list_backups.sh [hunk 22]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+if [ $# -gt 1 ]; then
```
Comment (jammsen) [inline scripts/list_backups.sh:3]: and here

File: scripts/rcon_cmd.sh [hunk 23]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# shellcheck disable=SC1091
+source /includes/rcon.sh
```
Comment (jammsen) [inline scripts/rcon_cmd.sh:4]: Why does this exist? We can simply call the rcon/rconcli binary.

File: includes/rcon.sh [hunk 24]
```diff
@@ -0,0 +1,25 @@
+# shellcheck disable=SC2148,SC1091
+source /includes/colors.sh
+
+# Function to run RCON commands
+# Arguments: <command>
+# Example: run_rcon "showplayers"
+
+run_rcon() {
```
Comment (jammsen) [inline includes/rcon.sh:8]: Why does this exist?

File: scripts/restart.sh [hunk 25]
```diff
@@ -0,0 +1 @@
+# Being implemented soon ...
```
Comment (jammsen) [inline scripts/restart.sh:1]: If you dont have anything, dont create empty files in git.

File: scripts/restore.sh [hunk 26]
```diff
@@ -0,0 +1 @@
+# Being implemented soon ...
```
Comment (jammsen) [inline scripts/restore.sh:1]: If you dont have anything, dont create empty files in git.

File: scripts/servermanager.sh [hunk 27]
```diff
@@ -0,0 +1,167 @@
...
+    if [[ -n $WEBHOOK_ENABLED ]] && [[ $WEBHOOK_ENABLED == "true" ]]; then
+        send_webhook_notification "$WEBHOOK_START_TITLE" "$WEBHOOK_START_DESCRIPTION" "$WEBHOOK_START_COLOR"
+    fi
+    ./PalServer.sh "$START_OPTIONS"
+}
+
+function stop_server() {
+    pp --warning ">>> Stopping server...\n\n"
+
+    rc 'broadcast Server_Shutdown_requested' "> Broadcasting server shutdown request..."
```
Comment (jammsen) [inline scripts/servermanager.sh:47]: -.-

File: scripts/servermanager.sh [hunk 28]
```diff
@@ -0,0 +1,167 @@
...
+    rc 'broadcast Saving...'
+    rc 'save'
+    rc 'broadcast Done...' "> Broadcasting server shutdown request..."
```
Comment (jammsen) [inline scripts/servermanager.sh:50]: error, no rc

File: scripts/servermanager.sh [hunk 29]
```diff
@@ -0,0 +1,167 @@
...
+	tail --pid="$(pidof PalServer-Linux-Test)" -f 2>/dev/null
+
+    if [[ -n $WEBHOOK_ENABLED ]] && [[ $WEBHOOK_ENABLED == "true" ]]; then
+        send_webhook_notification "$WEBHOOK_STOP_TITLE" "$WEBHOOK_STOP_DESCRIPTION" "$WEBHOOK_STOP_COLOR"
+    fi
+	
+    exit 143;
+}
+
+function install_server() {
```
Comment (jammsen) [inline scripts/servermanager.sh:67]: This was extracted on purpose, why is this back here?

File: scripts/servermanager.sh [hunk 30]
```diff
@@ -0,0 +1,167 @@
...
+    # Force a fresh install of all
+    pp --success "\n>>> Doing a fresh install of the gameserver...\n\n"
+    if [[ -n $WEBHOOK_ENABLED ]] && [[ $WEBHOOK_ENABLED == "true" ]]; then
+        send_webhook_notification "Installing server" "Server is beeing installed" "$WEBHOOK_INFO_COLOR"
+    fi
+    ${steamcmd_dir}/steamcmd.sh +force_install_dir "$GAME_ROOT" +login anonymous +app_update 2394010 validate +quit
+    pp --success ">>> Done installing the gameserver.\n"
+}
+
+function update_server() {
```
Comment (jammsen) [inline scripts/servermanager.sh:77]: Same for this

File: scripts/servermanager.sh [hunk 31]
```diff
@@ -0,0 +1,167 @@
...
+
+### Server Functions
+
+function start_server() {
+    # IF Bash extension used:
+    # https://stackoverflow.com/a/13864829
+    # https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02
+
+    pp --success "\n>>> Starting the gameserver\n"
+    cd "$GAME_ROOT" || exit
```
Comment (jammsen) [inline scripts/servermanager.sh:25]: You created that in dockerfile and entrypoint, if that would error the pipeline would have already failed. Why check it again?

File: scripts/servermanager.sh [hunk 32]
```diff
@@ -0,0 +1,167 @@
...
+
+    rc 'broadcast Server_Shutdown_requested' "> Broadcasting server shutdown request..."
+    rc 'broadcast Saving...'
+    rc 'save'
+    rc 'broadcast Done...' "> Broadcasting server shutdown request..."
+
+
+    pp --info "> Creating server backup before stopping server...\n"
+
+    backup --create
```
Comment (jammsen) [inline scripts/servermanager.sh:55]: Why create a backup here?! 
Thats very bad! With that insane amount of memory leaks, we already fighting hard for even parts of a second to not run in the 10 sec timeout of docker-compose down, which hard-kills the container after exactly 10 seconds. Why add un-needed complexity?
From a SW-Arch-POV:
The user wanted to to stop the server, where did he/she sign up to create a backup?

File: scripts/servermanager.sh [hunk 33]
```diff
@@ -0,0 +1,167 @@
...
+        pp --info "> Setting Multi-Core-Enhancements to enabled\n"
+        START_OPTIONS="$START_OPTIONS -useperfthreads -NoAsyncLoadingThread -UseMultithreadForDS"
+    fi
+    if [[ -n $WEBHOOK_ENABLED ]] && [[ $WEBHOOK_ENABLED == "true" ]]; then
+        send_webhook_notification "$WEBHOOK_START_TITLE" "$WEBHOOK_START_DESCRIPTION" "$WEBHOOK_START_COLOR"
+    fi
+    ./PalServer.sh "$START_OPTIONS"
+}
+
+function stop_server() {
```
Comment (jammsen) [inline scripts/servermanager.sh:44]: This function is so complex and brings unneeded complexity. The old version did what it should do. I dont understand why this exists.

File: scripts/servermanager.sh [hunk 34]
```diff
@@ -0,0 +1,167 @@
...
+
+
+### Signal Handling (Handlers & Traps)
+
+# Handler for SIGTERM from docker stop command
+function termHandler() {
+    stopServer
+}
+
+function start_handlers() {
```
Comment (jammsen) [inline scripts/servermanager.sh:105]: This function doesnt exist either, its litteraly a function around rainbow-puke to establish 1 line, that was before that in the right place. Why does this exist?

File: scripts/servermanager.sh [hunk 35]
```diff
@@ -0,0 +1,167 @@
...
+    # Start the server manager
+    start_main &
+
+    killpid="$!"
+    pp --info "\n>>> Server main thread started with pid ${killpid}\n"
+    wait ${killpid}
+    
+    # pp --warning ">>> Server stopped, checking for restart or backup restore\n"
+
+    # # If file 'restart' in server_restart exists, we do a server restart
```
Comment (jammsen) [inline scripts/servermanager.sh:152]: Why is this commented out?
Does it work?
Is it final?
Do i need to review it when its commented out?

**Review by jammsen — CHANGES_REQUESTED** [review]
Added my thoughts and comments.

**Review by thejcpalma — COMMENTED** [review]
Fixed all changes requested in review with @jammsen

File: scripts/servermanager.sh [hunk 36]
```diff
@@ -31,15 +31,53 @@ function start_server() {
         START_OPTIONS="$START_OPTIONS -useperfthreads -NoAsyncLoadingThread -UseMultithreadForDS"
     fi
     if [[ -n $WEBHOOK_ENABLED ]] && [[ $WEBHOOK_ENABLED == "true" ]]; then
-        send_webhook_notification "$WEBHOOK_START_TITLE" "$WEBHOOK_START_DESCRIPTION" "$WEBHOOK_START_COLOR"
+        send_start_notification
+        send_start_notification
```
Comment (thejcpalma) [inline scripts/servermanager.sh:35]: Fix duplication here

**Comment by thejcpalma** [comment]
Some explanations on some things:
- New line a the end of files -> https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file
- [`-passthrough-logs`](https://github.com/aptible/supercronic/pull/67) in supercronic -> Improves logging as it keeps supercronic job start and end and its status but improves readability of what the job prints (ignore rcon connection refused errors):
    ```bash
    2024-02-04 12:19:00 palworld-dedicated-server  | time="2024-02-04T12:19:00Z" level=info msg=starting iteration=1969 job.command="backup_manager --create" job.position=0 job.schedule="*/1 * * * *"
    2024-02-04 12:19:00 palworld-dedicated-server  | cli: execute: auth: rcon: dial tcp 127.0.0.1:25577: connect: connection refused
    2024-02-04 12:19:05 palworld-dedicated-server  | cli: execute: auth: rcon: dial tcp 127.0.0.1:25577: connect: connection refused
    2024-02-04 12:19:05 palworld-dedicated-server  | cli: execute: auth: rcon: dial tcp 127.0.0.1:25577: connect: connection refused
    2024-02-04 12:19:06 palworld-dedicated-server  | cli: execute: auth: rcon: dial tcp 127.0.0.1:25577: connect: connection refused
    2024-02-04 12:19:06 palworld-dedicated-server  | cli: execute: auth: rcon: dial tcp 127.0.0.1:25577: connect: connection refused
    2024-02-04 12:19:07 palworld-dedicated-server  | tar: Saved: Cannot stat: No such file or directory
    2024-02-04 12:19:07 palworld-dedicated-server  | tar: Exiting with failure status due to previous errors
    2024-02-04 12:19:08 palworld-dedicated-server  | cli: execute: auth: rcon: dial tcp 127.0.0.1:25577: connect: connection refused
    2024-02-04 12:19:08 palworld-dedicated-server  | >> Backup 'saved-20240204_121900.tar.gz' created successfully.
    2024-02-04 12:19:08 palworld-dedicated-server  | time="2024-02-04T12:19:08Z" level=info msg="job succeeded" iteration=1969 job.command="backup_manager --create" job.position=0 job.schedule="*/1 * * * *"
    ```
- `docs/` dir now has all new .md used to specify topics and help without cluttering the main `README.md`

Any questions just ask 😉

**Comment by jammsen** [comment]
As discussed #179  continues this PR and was just merged to develop. Closing this as agreed.
